packages/components/src/form-token-field/suggestions-list.js

```diff
@@ -2,157 +2,147 @@
  * External dependencies
  */
 import { map } from 'lodash';
-import scrollIntoView from 'dom-scroll-into-view';
+import scrollView from 'dom-scroll-into-view';
 import classnames from 'classnames';
 
 /**
  * WordPress dependencies
  */
-import { Component } from '@wordpress/element';
+import { useEffect, useState } from '@wordpress/element';
 import { withSafeTimeout } from '@wordpress/compose';
 
-class SuggestionsList extends Component {
-	constructor() {
-		super( ...arguments );
-		this.handleMouseDown = this.handleMouseDown.bind( this );
-		this.bindList = this.bindList.bind( this );
-	}
+function SuggestionsList( {
+	selectedIndex,
+	scrollIntoView,
+	match = '',
+	onHover = () => {},
+	onSelect = () => {},
+	suggestions = Object.freeze( [] ),
+	displayTransform,
+	instanceId,
+	setTimeout,
+} ) {
+	const [ scrollingIntoView, setScrollingIntoView ] = useState( true );
+	const [ list, setList ] = useState();
 
-	componentDidUpdate() {
+	useEffect( () => {
 		// only have to worry about scrolling selected suggestion into view
 		// when already expanded
 		if (
-			this.props.selectedIndex > -1 &&
-			this.props.scrollIntoView &&
-			this.list.children[ this.props.selectedIndex ]
+			selectedIndex > -1 &&
+			scrollIntoView &&
+			list.children[ selectedIndex ]
 		) {
-			this.scrollingIntoView = true;
-			scrollIntoView(
-				this.list.children[ this.props.selectedIndex ],
-				this.list,
-				{
-					onlyScrollIfNeeded: true,
-				}
-			);
+			setScrollingIntoView( true );
+			scrollView( list.children[ selectedIndex ], list, {
+				onlyScrollIfNeeded: true,
+			} );
 
-			this.props.setTimeout( () => {
-				this.scrollingIntoView = false;
+			setTimeout( () => {
+				setScrollingIntoView( false );
 			}, 100 );
 		}
-	}
+	}, [ selectedIndex, scrollIntoView, list ] );
 
-	bindList( ref ) {
-		this.list = ref;
-	}
+	const bindList = ( ref ) => {
+		setList( ref );
+	};
 
-	handleHover( suggestion ) {
+	const handleHover = ( suggestion ) => {
 		return () => {
-			if ( ! this.scrollingIntoView ) {
-				this.props.onHover( suggestion );
+			if ( ! scrollingIntoView ) {
+				onHover( suggestion );
 			}
 		};
-	}
+	};
 
-	handleClick( suggestion ) {
+	const handleClick = ( suggestion ) => {
 		return () => {
-			this.props.onSelect( suggestion );
+			onSelect( suggestion );
 		};
-	}
+	};
 
-	handleMouseDown( e ) {
+	const handleMouseDown = ( e ) => {
```

Comment:
This function could be moved to the top level of the file to avoid recreating it on every render.